### PR TITLE
Ensure prometheus metrics with the same name are serialized as a group.

### LINKED ISSUE #4382: Metric queueSize twice in Prometheus output
The prometheus endpoint produces invalid output.

The java application is started with:
`OTEL_METRICS_EXPORTER=prometheus`
`OTEL_TRACES_EXPORTER=otlp`
`OTEL_LOGS_EXPORTER=otlp`

After a while the produced output contains this information

``` 
# .... lines above this removed ....
# TYPE queueSize gauge
# HELP queueSize The number of logs queued
queueSize{logProcessorType="BatchLogProcessor"} 0.0 1649922810502
# TYPE processedLogs_total counter
# HELP processedLogs_total The number of logs processed by the BatchLogProcessor. [dropped=true if they were dropped due to high throughput]
processedLogs_total{dropped="false",logProcessorType="BatchLogProcessor"} 11.0 1649922810502
# TYPE runtime_jvm_gc_count_total counter
# HELP runtime_jvm_gc_count_total The number of collections that have occurred for a given JVM garbage collector.
runtime_jvm_gc_count_total{gc="Copy"} 230.0 1649922810502
runtime_jvm_gc_count_total{gc="MarkSweepCompact"} 13.0 1649922810502
# TYPE runtime_jvm_gc_time_total counter
# HELP runtime_jvm_gc_time_total Time spent in a given JVM garbage collector in milliseconds.
runtime_jvm_gc_time_total{gc="Copy"} 4150.0 1649922810502
runtime_jvm_gc_time_total{gc="MarkSweepCompact"} 6399.0 1649922810502
# TYPE otlp_exporter_seen_total counter
# HELP otlp_exporter_seen_total 
otlp_exporter_seen_total{type="log"} 11.0 1649922810502
otlp_exporter_seen_total{type="span"} 9371.0 1649922810502
# TYPE otlp_exporter_exported_total counter
# HELP otlp_exporter_exported_total 
otlp_exporter_exported_total{success="true",type="log"} 11.0 1649922810502
otlp_exporter_exported_total{success="true",type="span"} 9371.0 1649922810502
# TYPE processedSpans_total counter
# HELP processedSpans_total The number of spans processed by the BatchSpanProcessor. [dropped=true if they were dropped due to high throughput]
processedSpans_total{dropped="false",spanProcessorType="BatchSpanProcessor"} 9371.0 1649922810502
# TYPE queueSize gauge
# HELP queueSize The number of spans queued
queueSize{spanProcessorType="BatchSpanProcessor"} 1.0 1649922810502
```

We read the prometheus endpoint with Telegraf as we get this error:

`[inputs.prometheus] Error in plugin: error reading metrics for http://localhost:9088/metrics: reading text format failed: text format parsing error in line 115: second TYPE line for metric name "queueSize", or TYPE reported after samples`

queueSize is metric is twice in the output, once for logs and once for spans.

This should be grouped together, like this:

```
# TYPE queueSize gauge
# HELP queueSize The number of logs queued
queueSize{logProcessorType="BatchLogProcessor"} 0.0 1649922810502
queueSize{spanProcessorType="BatchSpanProcessor"} 1.0 1649922810502
```

But it is now appearing as 2 different metrics, which is not valid.

### PR DESCRIPTION
**Pull request by anuraaga**
This doesn't stop us from exporting two identical time series, filed https://github.com/open-telemetry/opentelemetry-specification/issues/2493 for spec on working on that. This should still help in many cases.

Fixes #4382

### CODE REVIEW

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/SerializerTest.java [hunk 1]
```diff
@@ -51,7 +52,10 @@ class SerializerTest {
               AggregationTemporality.CUMULATIVE,
               Collections.singletonList(
                   ImmutableDoublePointData.create(
-                      1633947011000000000L, 1633950672000000000L, KP_VP_ATTR, 5))));
+                      1633947011000000000L,
+                      1633950672000000000L,
+                      Attributes.of(TYPE, "mcds"),
```
Comment (anuraaga) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/SerializerTest.java:57]: Before all the metrics had the same name, meter, attributes, which is not representative of anything the SDK could actually output. So added this attribute to make a bit more realistic

File: exporters/prometheus/src/testJpms/java/io/opentelemetry/exporters/prometheus/test/JpmsTest.java [hunk 2]
```diff
@@ -14,7 +14,9 @@ class JpmsTest {
   @Test
   void noLinkageErrors() {
     SdkMeterProvider meterProvider =
-        SdkMeterProvider.builder().registerMetricReader(PrometheusHttpServer.create()).build();
+        SdkMeterProvider.builder()
+            .registerMetricReader(PrometheusHttpServer.builder().setPort(0).build())
```
Comment (anuraaga) [inline exporters/prometheus/src/testJpms/java/io/opentelemetry/exporters/prometheus/test/JpmsTest.java:18]: This test flaked on this PR so added the fix in

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/SerializerTest.java [hunk 3]
```diff
@@ -367,51 +366,25 @@ void openMetrics() {
...
-                + "instrument_name{kp=\"vp\",quantile=\"0.9\"} 0.1 1633950672.000\n"
-                + "instrument_name{kp=\"vp\",quantile=\"0.99\"} 0.3 1633950672.000\n"
-                + "# TYPE instrument_name histogram\n"
-                + "# HELP instrument_name description\n"
-                + "instrument_name_count{kp=\"vp\"} 2.0 1633950672.000\n"
-                + "instrument_name_sum{kp=\"vp\"} 1.0 1633950672.000\n"
-                + "instrument_name_bucket{kp=\"vp\",le=\"+Inf\"} 2.0 1633950672.000 # {span_id=\"0000000000000002\",trace_id=\"00000000000000000000000000000001\"} 4.0 0.001\n"
-                + "# TYPE instrument_name gauge\n"
-                + "# HELP instrument_name description\n"
+                + "instrument_name_total{type=\"mcds\"} 5.0 1633950672.000\n"
```
Comment (jack-berg) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/SerializerTest.java:369]: Not related to this specific PR, but the spec [requires](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#metric-metadata-1) that unit be appended to the metric name as a suffix. I don't believe we do that anywhere. I believe that would represent a breaking behavior change for prometheus users as their metric names would change when unit is present. 

Still, would be good to do that soon to minimize the impact.

**Review by jack-berg — APPROVED** [review]

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/SerializerTest.java (hunk 3 above)
Comment (anuraaga) [reply exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/SerializerTest.java:369]: Thanks for noticing this - filed https://github.com/open-telemetry/opentelemetry-specification/issues/2497

**Pull request merged**